src/Compression/CachedCompressedReadBuffer.cpp

```diff
@@ -87,7 +87,7 @@ CachedCompressedReadBuffer::CachedCompressedReadBuffer(
 void CachedCompressedReadBuffer::seek(size_t offset_in_compressed_file, size_t offset_in_decompressed_block)
 {
     /// Nothing to do if we already at required position
-    if (file_pos == offset_in_compressed_file
+    if (!owned_cell && file_pos == offset_in_compressed_file
```

Comment:
Without this change we could proceed to next compressed block and leave `offset` of buffer in inconsistent state. (when we have `offset() != offset_in_decompressed_block`, but `nextimpl_working_buffer_offset == offset_in_decompressed_block`, because `nextimpl_working_buffer_offset` is left from previous call.